superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -861,19 +925,98 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
     }
   }, [availableDbs]);
 
-  const tabChange = (key: string) => {
-    setTabKey(key);
+  const onDbImport = (info: UploadChangeParam) => {
+    setImportingModal(true);
+    setFile([
+      {
+        ...info.file,
+        status: 'done',
+      },
+    ]);
+
+    if (!(info.file.originFileObj instanceof File)) return;
+    importResource(info.file.originFileObj, passwords, confirmedOverwrite);
+  };
+
+  const passwordNeededField = () => {
+    if (passwordsNeeded.length === 0) return null;
```

Comment:
!passwordsNeed.length

Reply:
Ooh this looks nicer, thanks! Changed in [`this commit`](https://github.com/apache/superset/pull/19314/commits/aa5b7ea68452ec291ddbeaecfb73f2d4f312c6e3).